### ISSUE TITLE
What user information do we need to save?

### ISSUE THREAD

**Issue by Phavion** [svsticky/undead-mongoose#4]
As we want undead-mongoose to only obtain user information when needed from Koala, we only store the (uu)id of the registered users. I foresee some issues with this. For example, 'opwaarderen' would introduce an issue where the board member needs to find the user that needs the balance raised. Searching on the (uu)id would be very cumbersome. We could save first and last names but that is not guaranteed to be unique. Saving the student number could be an option but then the board member would need to keep asking for them.

tl;dr, for raising the balance, some user information needs to be stored to be able to search on by the board member. What information?

**Comment by Riscky** [comment]
I would say we need at least the UUID (to sync with Koala), name (for the board members convenience), student number (for linking cards) and age (for selling alcohol).
We need to figure out however how we want to sync changes and account deletions, I'm not sure how that is handled best.

**Comment by Phavion** [comment]
Either let koala send a push message, or pull every now and then from undead mongoose.
I would say the last as that reduces dependency on Koala.

**Comment by Riscky** [comment]
Maybe this would be a good occasion to implement webhooks in Koala.

**Comment by Phavion** [comment]
So you would say that we let Koala push when a user is updated. This would make the update instant but would it be an issue that you require the external system to implement this webhook? If we want undead-mongoose to be separate from 'koala', then pulling seems a better way to go as Koala would be passive.

**Comment by Phavion** [comment]
Furthermore, we do not have to save the studentnumber, I think. As long as we use the user_id from Koala.

**Comment by Riscky** [comment]
> The student number should only be needed when obtaining the user_id from koala the first time.

You're right, we don't have to save this at all.

**Comment by Riscky** [comment]
> So you would say that we let Koala push when a user is updated.

We could just send a webhook when a user is updated, where the webhook contains the relevant UUID of the updated user.
Then the receiving application (in this case `undead-mongoose`) can get the needed information from the Koala api.
I'm not advocating full fledged webhooks that contain all updated information, since that only tightens coupling between applications in my opinion, but this kind of minimal webhooks are quite useful to notify applications that resources are updated.

**Comment by Phavion** [comment]
Webhook communication implemented in c52bcef81ac7428f9ec14013ad1e0887fcfe2cc6
